File: lib/Models/CourseColorManager.dart
import 'package:flutter/material.dart';

class CourseColorManager {
  static Color generateSoftColor(key) {
    final hashCode = key.hashCode;
    final hue = (hashCode % 360).toDouble();
    const saturation = 0.4; // 低饱和度
    const lightness = 0.6; // 中等明度
    return HSLColor.fromAHSL(0.75, hue, saturation, lightness).toColor();
  }
}

File: android/app/src/main/kotlin/com/example/ios_club_app/CourseColorManager.kt
package com.example.ios_club_app

import android.graphics.Color

class CourseColorManager {
    companion object {
        /**
         * 根据课程名称生成柔和的颜色
         * 算法与 Flutter 端保持一致
         */
        fun generateSoftColor(key: String): Int {
            val hashCode = key.hashCode()
            val hue = (hashCode % 360).toDouble()
            val saturation = 0.4f
            val lightness = 0.6f
            val hsv = floatArrayOf(hue.toFloat(), saturation, lightness)
            return Color.HSVToColor(hsv)
        }
    }
}